dsgrid/registry/project_registry_manager.py

```diff
@@ -871,6 +877,37 @@ def _submit_dataset(
             project_config.model.project_id,
         )
 
+    def _check_dataset_time_interval_type(
+        self, project_config: ProjectConfig, dataset_config: DatasetConfig
+    ):
+        dtime = dataset_config.get_dimension(DimensionType.TIME)
+        dtime_interval = dtime.get_time_interval_type()
+        ptime = project_config.get_base_dimension(DimensionType.TIME)
+        ptime_interval = ptime.get_time_interval_type()
+
+        if dtime_interval != ptime_interval:
+            dtime_timestamps = dtime.list_expected_dataset_timestamps()
+
+            # convert time ranges to matching time_interval_type and see if they match
+            match (dtime_interval, ptime_interval):
+                case (TimeIntervalType.PERIOD_BEGINNING, TimeIntervalType.PERIOD_ENDING):
+                    dtime_timestamps = [
+                        DatetimeTimestampType(x.timestamp + dtime.get_frequency())
+                        for x in dtime.list_expected_dataset_timestamps()
+                    ]
+                case (TimeIntervalType.PERIOD_ENDING, TimeIntervalType.PERIOD_BEGINNING):
+                    dtime_timestamps = [
+                        DatetimeTimestampType(x.timestamp - dtime.get_frequency())
+                        for x in dtime.list_expected_dataset_timestamps()
+                    ]
```

Comment:
This is essentially a repeat of the logic in DatasetSchemaHandlerBase._convert_dataset_time_to_project_time_interval. Should this conversion logic be a utility somewhere with the time functionality?

Reply:
Same logic but completely different implementation because one is for list of time objects and the other for spark df. So not the easiest to factor out.